.github/workflows/go.yml

```
@@ -39,5 +39,32 @@ jobs:
 
+    - name: Build
+      run: go install
+
     - name: Test
-      run: go test -v ./...
+      run: |
+        go test -v ./... -coverprofile coverage.out -coverpkg ./...
+        go tool cover -func coverage.out -o coverage.out # Replaces coverage.out with the analysis of coverage.out
+    - name: Go Coverage Badge
+      uses: tj-actions/coverage-badge-go@v1
+      if: ${{ runner.os == 'Linux' && matrix.go == '1.22' }} # Runs this on only one of the ci builds.
```

actionlint

property "go" is not defined in object type {}

(expression)